src/App.jsx

```diff
@@ -1,23 +1,49 @@
+// import styles from './App.module.css'; /* CSS Modules */
+
 import './App.css'; /* Global CSS */
-import styles from './App.module.css'; /* CSS Modules */
+import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
+import Header from './components/Header/Header';
+import Home from './views/Home/Home';
+import Auth from './views/Auth/Auth';
+import ConfirmEmail from './views/ConfirmEmail/ConfirmEmail';
+import Profile from './views/Profile/Profile';
+import UpdateProfile from './views/UpdateProfile/UpdateProfile';
+import UserProvider from './context/UserContext';
+import ProfileProvider from './context/ProfileContext';
 
 export default function App() {
   return (
-    <h1
-      className={`
-      bg-green-400
-        text-3xl
-        text-center
-      text-white
-        font-bold
-        p-10
-        w-1/2
-        mx-auto
-        mt-10
-        ${styles.myCustomCssClass}
-      `}
-    >
-      Hello, World!
-    </h1>
+    <div className="App">
+      <UserProvider>
+        <Router>
+          <Header />
+          <Switch>
+            <Route exact path="/">
+              <Home />
+            </Route>
+            <Route path="/register">
+              <Auth isRegistering />
+            </Route>
+            <Route path="/login">
+              <Auth />
+            </Route>
+            <Route path="/confirm-email">
+              <ConfirmEmail />
+            </Route>
+            <ProfileProvider>
```

Comment:
It may make more sense to just create a custom hook for your profile without the as context is typically used for data that needs to be used throughout the entire application